torchinductor/dependencies.py

```diff
@@ -71,6 +80,10 @@ def store(self, name, index, value):
     def reduction(self, name, dtype, reduction_type, index, value):
         return self.store(name, index, f"reduce_{reduction_type})({value})")
 
+    def index_expr(self, expr, dtype):
```

Comment:
Curious, where is this called?

Reply:
`torch.arange`, `torch.triu`, padding, etc.  It is when there is data flow from indexing variables into compute.